docs/aws/audit/elbmonitoring/rules/elb_no_deletion_policy.mdx

```diff
@@ -24,6 +24,208 @@ HIPAA, NIST
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a JSON output with details of all the load balancers.
+
+3. Extract Load Balancer ARNs: From the JSON output, extract the ARN of each load balancer. You can use the following command to do this:
+
+   ```
+   aws elbv2 describe-load-balancers --query 'LoadBalancers[*].[LoadBalancerArn]' --output text
+   ```
+   This command will return a list of all the load balancer ARNs.
+
+4. Check Deletion Protection Flag: For each load balancer ARN, use the following command to check if the deletion protection flag is enabled:
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+   Replace `<load-balancer-arn>` with the ARN of the load balancer you want to check. This command will return a JSON output with the attributes of the load balancer. If the deletion protection flag is enabled, the `delete_protection.enabled` attribute will be set to `true`. If it's not, then the load balancer is misconfigured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install Boto3 in your environment. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials either by setting environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`, or by creating the credentials file at `~/.aws/credentials`.
+
+2. Import necessary libraries and create a session:
+   Import the Boto3 library in your Python script and create a session using your AWS credentials.
+   ```python
+   import boto3
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       aws_session_token='SESSION_TOKEN',
+   )
+   ```
+
+3. Get a list of all ELBs and check the deletion protection attribute:
+   Use the `describe_load_balancers` method to get a list of all ELBs. Then, for each ELB, use the `describe_load_balancer_attributes` method to get its attributes and check if the `deletion_protection.enabled` attribute is set to `True`.
+   ```python
+   elbv2 = session.client('elbv2')
+   elbs = elbv2.describe_load_balancers()
+   for elb in elbs['LoadBalancers']:
+       attributes = elbv2.describe_load_balancer_attributes(LoadBalancerArn=elb['LoadBalancerArn'])
+       for attr in attributes['Attributes']:
+           if attr['Key'] == 'deletion_protection.enabled':
+               print(f"ELB {elb['LoadBalancerName']} has deletion protection enabled: {attr['Value']}")
+   ```
+
+4. Analyze the output:
+   The script will print the name of each ELB and whether it has deletion protection enabled. If the value is `False`, it means the ELB does not have deletion protection enabled and is potentially misconfigured.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>